synapse/lib/storm.py

```diff
@@ -947,3 +948,65 @@ def getArgParser(self):
 
             if self.opts.join:
                 yield node, path
+
+class ScrapeCmd(Cmd):
+    '''
+    Use textual properties of existing nodes to find other easily recognizable nodes.
+
+    Examples:
+
+        inet:search:query | scrape
+        inet:search:query | scrape --refs
+    '''
+
+    name = 'scrape'
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+
+        pars.add_argument('--props', '-p', nargs='+', type=str, default=[],
+                          help='Specify relative properties to scrape')
+        pars.add_argument('--refs', '-r', default=False, action='store_true',
+                          help='Create refs to any scraped nodes from the source node')
+        pars.add_argument('-j', '--join', default=False, action='store_true',
+                          help='Include source nodes in the output of the command.')
+
+        return pars
+
+    async def execStormCmd(self, runt, genr):
+
+        async for node, path in genr:  # type: s_node.Node, s_node.Path
+
+            # repr all prop vals and try to scrape nodes from them
+            reprs = node.reprs()
+
+            props = [k for k in node.props.keys()]
+
+            # make sure any provided props are valid
+            for fprop in self.opts.props:
+                if fprop in props:
```

Comment:
`node.form.props` will check whether or not the props provided are valid for the form, vs. valid for a given instance of a form.

Reply:
This is really relevant since we're more interested in knowing the props available for a given form rather than the props existing on a given node.